### PR TITLE
Fix macOS aarch64 compile errors

### PR DESCRIPTION
**Pull request by chadbrewbaker**
Fixing macOS aarch64 compile errors. df was bringing in an x86 symbol and libc had to be bumped so 'cp' would link properly.

### CODE REVIEW

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ env:
   PROJECT_NAME: coreutils
   PROJECT_DESC: "Core universal (cross-platform) utilities"
   PROJECT_AUTH: "uutils"
-  RUST_MIN_SRV: "1.32.0" ## v1.32.0 - minimum version for half, tempfile, etc
+  RUST_MIN_SRV: "1.49.0" ## v1.33.0 - minimum version for aarch64-apple
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:14]: This is a huge bump.
I am ok jumping to 33  but 49 is a bit too much :/

File: src/uu/df/src/df.rs [hunk 2]
```diff
@@ -215,10 +215,7 @@ extern "C" {
     #[link_name = "getmntinfo$INODE64"]
     fn getmntinfo(mntbufp: *mut *mut statfs, flags: c_int) -> c_int;
 
-    #[cfg(all(
-        target_os = "freebsd",
-        not(all(target_os = "macos", target_arch = "x86_64"))
-    ))]
+    #[cfg(any( all(target_os = "freebsd"), all(target_os = "macos", target_arch="aarch64")))]
```
Comment (sylvestre) [inline src/uu/df/src/df.rs:218]: please run rustfmt

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (chadbrewbaker) [reply .github/workflows/CICD.yml:14]: 1.49 is the minimum version for macOS aarch64 support.  I think 1.33 covers the libc bump which is required even if you are using 1.49.

Comment (sylvestre) [reply .github/workflows/CICD.yml]: Sure but just create a new task to build it for aarch64-apple.
We can keep 32 for the rest of the platform

**Comment by chadbrewbaker** [comment]
Also muted the test that was breaking bash terminals. https://github.com/uutils/coreutils/issues/1719

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (chadbrewbaker) [reply .github/workflows/CICD.yml]: I just bumped it to 1.33 for the libc. Anyone compiling for aarch64-apple will be using 1.49+ anyway.

File: tests/by-util/test_tail.rs [hunk 3]
```diff
@@ -95,8 +95,8 @@ fn test_follow_stdin() {
         .stdout_is_fixture("follow_stdin.expected");
 }
 
-// FixME: test PASSES for usual windows builds, but fails for coverage testing builds (likely related to the specific RUSTFLAGS '-Zpanic_abort_tests -Cpanic=abort')
-#[cfg(not(windows))]
+// FixME: test PASSES for usual windows builds, but fails for coverage testing builds (likely related to the specific RUSTFLAGS '-Zpanic_abort_tests -Cpanic=abort')  This test also breaks tty settings under bash requiring a 'stty sane' or reset.
+#[cfg(disable_until_fixed)]
```
Comment (sylvestre) [inline tests/by-util/test_tail.rs:99]: I am confused, why this regressed ?
new version of the libc crate ?

Comment (sylvestre) [reply tests/by-util/test_tail.rs:99]: btw why disabling it everywhere ? Linux should still work ?

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (sylvestre) [reply .github/workflows/CICD.yml]: Is the bump of 1.33 still necessary?

File: tests/by-util/test_tail.rs (hunk 3 above)
Comment (chadbrewbaker) [reply tests/by-util/test_tail.rs:99]: Orthogonal issue see  https://github.com/uutils/coreutils/issues/1719 . 'cargo test' breaks the bash terminal when this runs. You have to either 'stty sane' or reset.

Comment (chadbrewbaker) [reply tests/by-util/test_tail.rs:99]: If I recall it didn't break the terminal under zsh. I didn't try other shells.

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (sylvestre) [reply .github/workflows/CICD.yml]: Is dumping 1.33 necessary for this PR ? thanks and sorry to be a pain :)

Comment (chadbrewbaker) [reply .github/workflows/CICD.yml]: libc, which is needed on the M1 for 'cp' to link, has a minimum version of 1.33 for all dependencies https://crates.io/crates/libc

File: tests/by-util/test_tail.rs (hunk 3 above)
Comment (sylvestre) [reply tests/by-util/test_tail.rs:99]: ok, I am using zsh, this is probably why I don't see it.

Could you please do a separate PR for this ?
I am not a big fan of mixing changes in the same PR

File: .github/workflows/CICD.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ env:
   PROJECT_NAME: coreutils
   PROJECT_DESC: "Core universal (cross-platform) utilities"
   PROJECT_AUTH: "uutils"
-  RUST_MIN_SRV: "1.32.0" ## v1.32.0 - minimum version for half, tempfile, etc
+  RUST_MIN_SRV: "1.33.0" ## v1.33.0 - minimum version for libc required on aarch64-apple
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:14]: OK, could you please update the doc too https://github.com/uutils/coreutils/blob/master/README.md#rust-version ?

Comment (sylvestre) [reply .github/workflows/CICD.yml:14]: half and tempfile are probably requiring something more recent than 1.33, no?
if so, could you please put back the comment

**Comment by chadbrewbaker** [comment]
latest tempfile 3.1.0 is min version 1.40.0,  https://github.com/uutils/coreutils/blob/master/Cargo.toml#L345 latest half is still 1.32.  Did you want me to bump it to 1.40.0 for tempfile?

**Comment by sylvestre** [comment]
@chadbrewbaker No, we should update when we have a real need :)

**Comment by chadbrewbaker** [comment]
Sorry didn't read your comment in time. Bumped to 1.40.0 since we already had comments in mainline Cargo.toml about 1.40.0 minimum.  Perhaps as another PR bisect back to the first version that doesn't break the tests and add that to the CI file and the Readme saying  the tests are passing back to min version X?

**Comment by chadbrewbaker** [comment]
So should I keep this at 1.40.0 or pin everything to 1.33.0 and note that tempfile might not be fully supported?

**Comment by sylvestre** [comment]
The reference about 1.40 is for tempfile 3.2 as comment for the future.
For now, we are relying on 3.1. So, please keep 1.33 as minimal version.

**Comment by chadbrewbaker** [comment]
Done.

**Pull request merged**